### PR TITLE
Add Container and Header components

### PR DESCRIPTION
**Pull request by nicmalegre**
## Add Container and Header components

---

#### Description:

* Add container component to use it in different screen and to be able to have SafeArea and Header in the same component.
* Add Header component following the figma desing of the app. It will allow show the screen title and also have left and right icon to execute different function on press like navigate back, show profile, open chat. 
* Add icons to use in the Header

---

#### Figma:

 

---

#### Screenshot: 

* Android (Pixel 3A):
 

 
* IOS (iPhone 13):

### CODE REVIEW

File: src/components/common/Header/styles.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    height: 60,
+    backgroundColor: WHITE,
+  },
+  icon: {
+    marginHorizontal: 16,
+    marginVertical: 8,
+    marginRight: 10,
```
Comment (santiagorobra) [inline src/components/common/Header/styles.js:16]: choose one, they are not cumulative
marginHorizontal: 16 or marginRight: 10?

File: src/components/common/Container/styles.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { WHITE } from 'constants/colors';
```
Comment (santiagorobra) [inline src/components/common/Container/styles.js:1]:
```suggestion
import { StyleSheet } from 'react-native';
import { WHITE } from 'constants/colors';
```

File: src/components/common/Header/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { WHITE } from 'constants/colors';
```
Comment (santiagorobra) [inline src/components/common/Header/index.js:1]:
```suggestion
import React from 'react';
import { View, Text, TouchableHighlight, Image } from 'react-native';
import { WHITE } from 'constants/colors';
import { CHAT_ICON, BACK_ARROW_ICON } from 'constants/icons';
import styles from './styles';
```

File: src/components/common/Header/styles.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { BLACK, WHITE } from 'constants/colors';
```
Comment (santiagorobra) [inline src/components/common/Header/styles.js:1]:
```suggestion
import { StyleSheet } from 'react-native';
import { BLACK, WHITE } from 'constants/colors';
import { PRIMARY_FONT_SEMI_BOLD } from 'constants/fonts';
```

File: src/screens/MainScreen/index.js [hunk 5]
```diff
@@ -6,6 +6,8 @@ import { logout } from 'actions/userActions';
 import { MAIN_SCREEN } from 'constants/screens';
```
Comment (santiagorobra) [inline src/screens/MainScreen/index.js:6]:
```suggestion
import strings from 'localization';
import useSession from 'hooks/useSession';
import Container from 'components/common/Container';
import { PROFILE_ICON } from 'constants/icons';
import { MAIN_SCREEN } from 'constants/screens';
import styles from './styles';
```

File: src/components/common/Container/index.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { SafeAreaView } from 'react-native';
+import Header from '../Header';
+import styles from './styles';
+
+const Container = ({ children, headerProps }) => {
+  const { testID, title, imageLeft, onPressIconLeft, imageRight, onPressIconRight } =
+    headerProps || {};
```
Comment (santiagorobra) [inline src/components/common/Container/index.js:8]:
```suggestion
```

File: src/components/common/Header/index.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+          style={styles.containerIconRight}>
+          <View style={styles.icon}>
+            <Image source={imageRight} />
+          </View>
+        </TouchableHighlight>
+      )}
+    </View>
+  );
+};
+
```
Comment (santiagorobra) [inline src/components/common/Header/index.js:45]: do not forget the propTypes

**Review by santiagorobra — APPROVED** [review]

**Comment by santiagorobra** [comment]
great job! 👏 👏 👏

**Review by asdolo — APPROVED** [review]

**Pull request merged**